### PR TITLE
feat: `inv_mod` implementation and coverage

### LINKED ISSUE #19: [Feature]: `inv_mod` function
Implement a function that calculate the modular multiplicative inverse of a number in `Z/nZ`.

Check out [Solidity Math Library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/math/Math.sol) as a reference.

### PR DESCRIPTION
**Pull request by ericnordelo**
Resolves #19  

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [ ] Changelog

## Release Notes

* **New Features**
  * Added modular arithmetic operations: `inv_mod()` for computing modular inverses and `mul_mod()` for modular multiplication across all integer types (u8, u16, u32, u64, u128, u256).

* **Bug Fixes**
  * Fixed arithmetic initialization for bit-shift operations to ensure consistent type handling.

* **Tests**
  * Added comprehensive test coverage for new modular arithmetic functionality, including edge cases and zero-modulus validation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: math/core/sources/internal/macros.move [hunk 1]
```diff
@@ -676,3 +716,83 @@ public(package) fun round_sqrt_result(
...
+/// Falls back to the wide (`u512`) helper when the operands exceed 128 bits so overflow cannot
+/// occur.
+public(package) fun mul_mod_impl(a: u256, b: u256, modulus: u256): u256 {
+    assert!(modulus != 0, EZeroModulus);
+    if (a == 0 || b == 0) {
+        return 0
+    };
+    if ((a >> 128) == 0 && (b >> 128) == 0) {
+        return ((a * b) % modulus)
+    };
```
Comment (immrsd) [inline math/core/sources/internal/macros.move]: We already perform this very check in `mul_div_inner`, `mul_shr_inner`, and `log2_should_round_up` that compares the values to the max `u128` value rather than shifting. It might be better to keep all the checks consistent
```
let max_small = std::u128::max_value!() as u256;
if (a > max_small || b > max_small) {
    // wide
} else {
    // fast
}
```

File: math/core/tests/macros_tests.move [hunk 2]
```diff
@@ -200,6 +200,94 @@ fun mul_div_wide_detects_overflowing_quotient() {
...
+fun mod_sub_impl_wraps_underflow() {
+    let result = macros::mod_sub_impl(3, 5, 11);
+    assert_eq!(result, 9);
+}
+
+#[test]
+fun mul_mod_impl_uses_fast_path_when_small() {
+    let result = macros::mul_mod_impl(7, 9, 11);
+    assert_eq!(result, 8);
+}
```
Comment (immrsd) [inline math/core/tests/macros_tests.move]: How does the check in this test case work?

Comment (immrsd) [reply math/core/tests/macros_tests.move]: Removed the test case
